test/create.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+var throws = require('assert').throws;
+
+var R = require('ramda');
+var $ = require('sanctuary-def');
+
+var eq = require('./utils').eq;
+var errorEq = require('./utils').errorEq;
+var S = require('..');
+
+
+//  customEnv :: Array Type
+var customEnv = S.env.concat([$.EnumType(['foo', true, 42])]);
+
+var checkedDefaultEnv   = S.create({checkTypes: true, env: S.env});
+var checkedCustomEnv    = S.create({checkTypes: true, env: customEnv});
+var uncheckedDefaultEnv = S.create({checkTypes: false, env: S.env});
+var uncheckedCustomEnv  = S.create({checkTypes: false, env: customEnv});
+
+
+describe('create', function() {
+
+  it('is a unary function', function() {
+    eq(typeof S.create, 'function');
+    eq(S.create.length, 1);
+  });
+
+  it('type checks its arguments', function() {
+    throws(function() { S.create({}); },
+           errorEq(TypeError,
+                   'Invalid value\n' +
+                   '\n' +
+                   'create :: { checkTypes :: Boolean, env :: Array Any } -> Object\n' +
+                   '          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n' +
+                   '                               1\n' +
+                   '\n' +
+                   '1)  {} :: Object, StrMap ???\n' +
+                   '\n' +
+                   'The value at position 1 is not a member of ‘{ checkTypes :: Boolean, env :: Array Any }’.\n'));
+
+    throws(function() { S.create({checkTypes: 'true', env: []}); },
+           errorEq(TypeError,
+                   'Invalid value\n' +
+                   '\n' +
+                   'create :: { checkTypes :: Boolean, env :: Array Any } -> Object\n' +
+                   '                          ^^^^^^^\n' +
+                   '                             1\n' +
+                   '\n' +
+                   '1)  "true" :: String\n' +
+                   '\n' +
+                   'The value at position 1 is not a member of ‘Boolean’.\n'));
+  });
+
+  it('returns a Sanctuary module', function() {
+    var expected = R.keys(S).sort();
+    eq(R.keys(checkedDefaultEnv).sort(), expected);
```

Comment:
We can't actually use `S.keys` here, since `S` is not a member of `StrMap a`. I've replaced `R.keys` with `Object.keys` (which reads nicely for getting the names of a record's fields).